servo/utilities/subprocess.py

```diff
@@ -266,6 +266,10 @@ async def stream_subprocess_shell(
             f"Subprocess finished with return code {result} in {duration} (`{cmd}`)"
         )
         return result
+    except asyncio.CancelledError:
+        # when used in asynccontextmanager, the wrapping task is cancelled after yield (aexit), cleanup the subprocess in that case
+        process.terminate()
+        raise
```

Comment:
Good catch here but I went with a slightly different spin on the handling: 76e6d2d47d95e0a6bbabbae76e75b2a108f0fdc1